flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -990,6 +991,10 @@ public MailboxExecutorFactory getMailboxExecutorFactory() {
         return this.mailboxProcessor::getMailboxExecutor;
     }
 
+    public Supplier<Boolean> getMailboxHasMail() {
+        return this.mailboxProcessor::hasMail;
```

Comment:
With this change, the `@VisibleForTesting` annotation can be removed for `MailboxProcessor#hasMail`.

Reply:
Thanks for catching this. It is fixed now.